go-controller/pkg/ovn/pods.go

```diff
@@ -120,12 +120,53 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod, portInfo *lpInfo) (err er
 		podIfAddrs = portInfo.ips
 	}
 
-	var allOps, ops []ovsdb.Operation
-	if ops, err = oc.deletePodFromNamespace(pod.Namespace, podIfAddrs, portUUID); err != nil {
-		return fmt.Errorf("unable to delete pod %s from namespace: %w", podDesc, err)
+	shouldRelease := true
+	// check to make sure no other pods are using this IP before we try to release it if this is a completed pod.
+	if util.PodCompleted(pod) {
+		if shouldRelease, err = oc.lsManager.ConditionalIPRelease(pod.Spec.NodeName, podIfAddrs, func() (bool, error) {
+			pods, err := oc.watchFactory.GetAllPods()
```

Comment:
is it easy to add `GetAllPodsOnNode(pod.Spec.NodeName)`? Then we don't need to iterate over all pods...although I admit I am not aware of how the PodLister works and if internally effect is the same,,,

Reply:
I tried to do that, but it looked like the API only exposed a way to filter by labels and not the  node.